packages/transformers/coffeescript/src/CoffeeScriptTransformer.js

```diff
@@ -1,11 +1,34 @@
+// @flow
 import {Transformer} from '@parcel/plugin';
+import SourceMap from '@parcel/source-map';
 import coffee from 'coffeescript';
+import {relativeUrl} from '@parcel/utils';
 
 export default new Transformer({
-  async transform({asset}) {
+  async transform({asset, options}) {
+    let sourceFileName: string = relativeUrl(
+      options.projectRoot,
+      asset.filePath
+    );
+
     asset.type = 'js';
-    let output = coffee.compile(await asset.getCode());
-    asset.setCode(output);
+    let output = coffee.compile(await asset.getCode(), {
+      filename: sourceFileName,
+      sourceMap: options.sourceMaps
+    });
+
+    // return from compile is based on sourceMaps option
+    if (options.sourceMaps) {
+      asset.setCode(output.js);
+      asset.setMap(
+        new SourceMap(output.sourceMap, {
```

Comment:
Shouldn't this be: `SourceMap.fromRawSourceMap(output.sourceMap)` ?

new SourceMap takes in an array of mappings. Not sure if `output.sourceMap` is an array of mappings or an actual sourceMap in case it is it should be as listed above.

Reply:
Oh good to know. I was basing it off of BabelTransformer, and didn't have much examples go to off.

Reply:
Not sure if there's any sourcemap tests for coffeescript. You can write some and base it of the JS sourcemap tests

Reply:
I mainly want to ensure they are being produced. I found the SourceMaps tests, but nothing that tested the maps file was generated or inlined.

Reply:
Basically, I have no faith in myself and unless I see a test that tells me it worked I won't believe I did it right.

Reply:
We should probably add one. For now you could just log whatever `output.sourceMap` is and based on that figure out what you should use. We can do the tests in a follow-up PR if that's easier

Reply:
Sounds good! Thanks

Reply:
Logging it showed me that it didn't match the contract. I knew it CS was outputting another map, so I checked that and it matched up. This should be all updated and set to go.